accounting/gst.js

```diff
@@ -150,11 +189,79 @@ async function generateB2bData(invoices) {
 }
 
 async function generateB2clData(invoices) {
-  return [];
+  const b2cl = [];
+
+  invoices.forEach(async (invoice) => {
+
+    const stateInvoiceRecord = {
+      pos: posMap[invoice.place.toUpperCase()],
+      inv: []
+    };
+
+    const invRecord = {
+      inum: invoice.invNo,
+      idt: DateTime.fromFormat(invoice.invDate, 'yyyy-MM-dd').toFormat(
+        'dd-MM-yyyy'
+      ),
+      val: invoice.invAmt,
+      itms: [],
+    }
+
+    let items = await frappe.db
+      .knex('SalesInvoiceItem')
+      .where('parent', invRecord.inum);
+
+    items.forEach((item) => {
+      const itemRecord = {
+        num: item.hsnCode || 0,
```

Comment:
@piyushsinghania why 0? I don't think we should default to any value. Let the field be null and the json validator will error out and point to the problem.

Reply:
Done 💯